### PR TITLE
add `into_ptr` to consume the wrapper

### PR DESCRIPTION
**Pull request by flier**
Maybe the name `into_raw` or `into_inner` will be better.

### CODE REVIEW

File: foreign-types-shared/src/lib.rs [hunk 1]
```diff
@@ -25,6 +25,9 @@ pub trait ForeignType: Sized {
 
     /// Returns a raw pointer to the wrapped value.
     fn as_ptr(&self) -> *mut Self::CType;
+
+    /// Consumes the wrapper and returnes the raw pointer
+    fn into_ptr(self) -> *mut Self::CType;
```
Comment (sfackler) [inline foreign-types-shared/src/lib.rs]: We should be able to implement this here rather than in the macro, which will avoid breaking back compat as well.

Comment (flier) [reply foreign-types-shared/src/lib.rs]: sure, I think we could extract the `into_ptr` to a standalone `IntoForeignType` trait.

Comment (sfackler) [reply foreign-types-shared/src/lib.rs]: It doesn't need to be a separate trait. You can just define the new method as

```rust
fn into_ptr(self) -> *mut Self::CType {
    let ptr = self.as_ptr();
    mem::forget(self);
    ptr
}
```

Comment (flier) [reply foreign-types-shared/src/lib.rs]: sorry, it seems I misunderstand your comment, thanks

**Comment by sfackler** [comment]
Thanks!

**Pull request merged**